gc.ts

```diff
@@ -380,20 +381,63 @@ export class MnS<A extends MarkableAllocator> {
         }
 
         case TAG_LIST: {
-          throw new Error("TODO: trace list");
-        }
+          // Layout: [32-bit TAG_LIST, 32-bit <length>, 32-bit <capacity>, data...]
+
+          // Extract value at childPtr + 4. Assumed to be a primitive value
+          const listLength = childSize;
+          
+          // Sanity check, just-in-case
+          // NOTE(sagar): probably not necessary
+          if(isPointer(listLength)) {
+            throw new Error("Pointer value stored in the place of list length");
+          }
 
-        case TAG_STRING: {
-          throw new Error("TODO: trace string");
-        }
+          // Note(sagar): Memory layout is abstracted by allocator
+          // childPtr always points to start of data, not header
+          for(let dataPtr = childPtr; dataPtr !== childPtr + listLength * 4n; dataPtr += 4n) {
+            const elementValue = this.getField(dataPtr);
+
+            if(isPointer(elementValue)) {
+              const fieldPointerValue = extractPointer(elementValue);
+              if(!this.isMarked(fieldPointerValue)) {
```

Comment:
`None` is represented as `i32.const 0`. Need to insert a check so we don't accidentally trace a bad pointer.

Reply:
Also, can you add that check for tracing fields with `TAG_CLASS` ?